pepys_import/core/store/common_db.py

```diff
@@ -558,6 +524,10 @@ def sensor(self):
     def sensor_name(self):
         return association_proxy("sensor", "name")
 
+    @declared_attr
+    def platform_id(self):
+        return association_proxy("sensor", "host")
+
```

Comment:
@robintw - can you explain the role/functionality of `association_proxy` to me please?

Reply:
Sure, it provides a way of getting a 'fake' attribute on a SQLAlchemy model that actually comes from a relationship. So, for this example we have a SQLAlchemy model that represents a row in the `States` table. This has a SQLAlchemy relationship defined which links the `State` entry to the `Sensor` entry that it refers to (via the `sensor_id` field). This association proxy basically just gives a nice name to an attribute that is on the other end of a relationship. In this case, the relationship is the Sensor relationship, and here we're giving a nice name to the `host` property of the sensor found through that relationship.

Basically, it's a nicer way of saying `State.sensor.host` - instead, you can just do `State.platform_id`.